### PR TITLE
reconstructure PostTaskButton, change the mechanism of active TaskD...

### PR DESCRIPTION
**Pull request by Changhong-st**
Achieved:
 1. reconstructure PostTaskButton
 2. change the mechanism of active TaskDescription errorHint to make it more hint friendly
 3. use short-circuit evaluation instead of conditional operator to make it more readable
 4. make single responsibility sense of TaskDescription in top-page component

Continuing: 
 1. add task location feature
 2. add close tag pop-up window
 3. add login-in feature
 4. add get quotes route
 5. reconstructure other component to achieve single responsibility and react thinking

### CODE REVIEW

File: src/pages/PostTask/Button/Button.jsx [hunk 1]
```diff
@@ -1,40 +1,31 @@
 import React from 'react';
-import styles from './Button.module.scss';
 
+import styles from './Button.module.scss';
 
 class Button extends React.Component {
   constructor(props){
```
Comment (JavaScriptN0ob) [inline src/pages/PostTask/Button/Button.jsx]: Button为什么是一个class组件呀

File: src/pages/PostTask/PostTaskForm/PostTaskForm.jsx [hunk 2]
```diff
@@ -106,14 +131,16 @@ class PostTaskForm extends React.Component {
...
+            jobTitle={this.state.jobTitle}
+            jobDetails={this.state.jobDetails}
+            jobTitleLength={this.state.jobTitleLength}
+            jobDetailsLength={this.state.jobDetailsLength}
+            onJobTitle={this.onJobTitle}
+            onJobDetails={this.onJobDetails}
+            handleJobTitle={this.handleJobTitle}
+            handleJobDetails={this.handleJobDetails}
+            handleNextClick={this.handleTaskDescriptionNextClick}
+            handleBackClick={this.handleBackClick}
```
Comment (JavaScriptN0ob) [inline src/pages/PostTask/PostTaskForm/PostTaskForm.jsx:161]: 解构 你懂得

File: src/pages/PostTask/PostTaskForm/TaskDescription/TaskInput/TaskInput.jsx [hunk 3]
```diff
@@ -14,25 +14,27 @@ function TaskInput({
...
         {taskInputHint}
       </label>
       <textarea 
         className={cx(
           {
-            'JobTitle': size === 'small',
-            'JobDetails': size === 'large',
+            'job_title': size === 'small',
+            'job_details': size === 'large',
           }
```
Comment (JavaScriptN0ob) [inline src/pages/PostTask/PostTaskForm/TaskDescription/TaskInput/TaskInput.jsx]: Key Value pair里面的key外面的引号非必须，然后建议双引

File: src/pages/PostTask/Button/Button.jsx (hunk 1 above)
Comment (KieraDOG) [reply src/pages/PostTask/Button/Button.jsx]: 这个 Button 责任不够清晰. 你去看 antd 或者 material 的 Button.都不会参杂着 Next 或者 Back 的逻辑

File: src/pages/PostTask/Button/Button.module.scss [hunk 4]
```diff
@@ -1,4 +1,4 @@
-.Button {
+.button {
```
Comment (KieraDOG) [inline src/pages/PostTask/Button/Button.module.scss:1]: 👍

File: src/pages/PostTask/PostTaskForm/TaskBudget/BudgetInput/BudgetInput.jsx [hunk 5]
```diff
@@ -2,30 +2,56 @@ import React from 'react';
...
   }
-  //根据公司要求开发一个输入组件，能安全没有后门去筛选用户输入的数据
-  //我考虑到复用性（reuseable），单一原则，readable
-  //用什么技术 实现 什么功能
-  //regex技术，solid原则封装组件，实现数字输入框复用性的component
-  //实现思路，js，key（A
-  //在数据库那边再做一个valid，double valid，避免绕过前端，call api
-  //如果前端不做，也可以，但是前端做了这样一个valid，可以很好节省性能
+
+  const errorHint = "Please suggest a budget between $" + minBudget + " and $" + maxBudget + " for your task";
```
Comment (KieraDOG) [inline src/pages/PostTask/PostTaskForm/TaskBudget/BudgetInput/BudgetInput.jsx]: 这里用 template string 好一点吧

File: src/pages/PostTask/PostTaskForm/TaskBudget/BudgetInput/BudgetInput.jsx [hunk 6]
```diff
@@ -2,30 +2,56 @@ import React from 'react';
...
+              onChange={onBudgetHour}
+              onInput={validateInput}
+            />
+            <div className={styles.radio}> /hrs </div>
+          </React.Fragment>
+          :
+        null}
+      </div>
+      {taskBudget < minBudget || taskBudget > maxBudget
+        ?
```
Comment (KieraDOG) [inline src/pages/PostTask/PostTaskForm/TaskBudget/BudgetInput/BudgetInput.jsx]: 这里用短路计算好一些

**Review by ttcode10 — APPROVED** [review]
Approved

File: src/pages/PostTask/PostTaskForm/TaskBudget/BudgetInput/BudgetInput.jsx (hunk 5 above)
Comment (JavaScriptN0ob) [reply src/pages/PostTask/PostTaskForm/TaskBudget/BudgetInput/BudgetInput.jsx]: const errorHint = "Please suggest a budget between $" + minBudget + " and $" + maxBudget + " for your task";

建议按照龙哥思路更换为下面的写法。
系统学习链接:https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

const errorHint = \`Please suggest a budget between &#36; ${minBudget} and ${maxBudget} for your task`\;

File: src/pages/PostTask/PostTaskForm/TaskBudget/BudgetInput/BudgetInput.jsx (hunk 6 above)
Comment (JavaScriptN0ob) [reply src/pages/PostTask/PostTaskForm/TaskBudget/BudgetInput/BudgetInput.jsx]: {(taskBudget < minBudget || taskBudget > maxBudget) && (<ErrorHint>{errorHint}</ErrorHint>)

File: src/pages/PostTask/PostTaskForm/PostTaskForm.jsx (hunk 2 above)
Comment (JavaScriptN0ob) [reply src/pages/PostTask/PostTaskForm/PostTaskForm.jsx:161]: 除了解构，
这个地方着重看一下：https://jxnblk.com/blog/defining-component-apis-in-react/ 
里面的Split components at data boundaries 这个章节

**Review by hachilam92 — APPROVED** [review]

**Pull request merged**